include/vkwrap/vkwrap/mman.h

```diff
@@ -0,0 +1,556 @@
+        if ( result != VK_SUCCESS )
+        {
+            throw vk::SystemError{ vk::Result{ result }, "Mman: buffer mapping error." };
+        }
+
+        return mapped;
+    } // map
+
+    void unmap( vk::Buffer buffer ) { vmaUnmapMemory( m_vma, *findAllocation( buffer ) ); }
+
+    void flush( vk::Buffer buffer )
+    {
+        VkResult result = vmaFlushAllocation( m_vma, *findAllocation( buffer ), 0, VK_WHOLE_SIZE );
+
+        if ( result != VK_SUCCESS )
+        {
+            throw vk::SystemError{ vk::Result{ result }, "Mman: buffer flushing error." };
+        }
+    } // flush
+
+    void copy(
+        vk::Buffer src_buffer,
+        vk::Buffer dst_buffer,
+        vk::DeviceSize src_offset,
+        vk::DeviceSize dst_offset,
+        vk::DeviceSize size )
+    {
+        vk::BufferCopy region{ src_offset, dst_offset, size };
+
+        OneTimeCommand cmd = createCommand();
+        cmd->copyBuffer( src_buffer, dst_buffer, std::array{ region } );
+        cmd.submitAndWait();
+    } // copy
+
+    void copy( vk::Buffer src_buffer, vk::Buffer dst_buffer )
+    {
+        copy( src_buffer, dst_buffer, 0, 0, findInfo( src_buffer )->size );
+    } // copy
+
+    void copy( vk::Buffer src_buffer, vk::Image dst_image )
+    {
+        // TODO: add interface to configure image and buffer params.
+
+        ImageInfo* image_info = findInfo( dst_image );
+        vk::BufferImageCopy region{};
+
+        region.bufferOffset = 0;
+        region.bufferRowLength = 0;
+        region.bufferImageHeight = 0;
+
+        region.imageSubresource.aspectMask = vk::ImageAspectFlagBits::eColor;
+
+        // We use images with one layer and one mipmap.
+        region.imageSubresource.mipLevel = 0;
+        region.imageSubresource.baseArrayLayer = 0;
+        region.imageSubresource.layerCount = 1;
+
+        region.imageOffset = vk::Offset3D{ 0, 0, 0 };
+        region.imageExtent = vk::Extent3D{
+            image_info->extent.width,
+            image_info->extent.height,
+            1 /* depth */
+        };
+
+        OneTimeCommand cmd = createCommand();
+        cmd->copyBufferToImage( src_buffer, dst_image, image_info->layout, std::array{ region } );
+        cmd.submitAndWait();
+    } // copy
+
+    void transit( vk::Image image, vk::ImageLayout new_layout )
+    {
+        ImageInfo* image_info = findInfo( image );
+
+        vk::ImageMemoryBarrier barrier{};
+        vk::PipelineStageFlags src_stage{};
+        vk::PipelineStageFlags dst_stage{};
+
+        try
+        {
+            barrier = createImageBarrierInfo( image, image_info->format, image_info->layout, new_layout );
+
+            auto [ src_stage_tmp, dst_stage_tmp ] = choosePipelineStages( image_info->layout, new_layout );
+            src_stage = src_stage_tmp;
+            dst_stage = dst_stage_tmp;
+        } catch ( Error& e )
+        {
+            throw Error{ std::string{ "Mman: " } + e.what() };
+        }
```

Comment:
This try/catch block is bug prone. What if an exception is thrown that does not bind to ```Error &```? Then this is a no-op. The only thing that it does is add a prefix to the error message + it reduces code readability. It think this try/catch should be removed

Reply:
There cannot be any other than `Error &` exception. Adding **Mman: ** tells you which vkwrap module called the function. It is a good help in debug.

But actually the function is called only in `Mman::transit`, so, it won't be hard to find out caller. I've removed it.